src/MoBi.Presentation/Presenter/ChartPresenter.cs

```diff
@@ -190,54 +207,76 @@ protected virtual void OnDragDrop(object sender, DragEventArgs e)
       }
 
       protected abstract bool CanDropSimulation { get; }
 
       protected abstract void MarkChartOwnerAsChanged();
 
-      private void addHistoricalResults(DataRepository repository)
+      private void addHistoricalResults(IReadOnlyList<DataRepository> repositories)
       {
-         addDataRepositorySimulation(repository);
-         _editorPresenter.AddDataRepository(repository);
-         repository.SetPersistable(persistable: true);
+         addDataRepositoriesToSimulation(repositories);
+         _editorPresenter.AddDataRepositories(repositories);
+         repositories.Each(repository => repository.SetPersistable(persistable: true));
       }
 
-      private void addObservedData(DataRepository repository)
+      private void addObservedData(IReadOnlyList<DataRepository> repositories)
       {
-         _editorPresenter.AddDataRepository(repository);
-         repository.Where(c => c.DataInfo.Origin == ColumnOrigins.Observation)
-            .Each(c => _editorPresenter.AddCurveForColumn(c.Id));
+         _editorPresenter.AddDataRepositories(repositories);
+         using (_chartUpdater.UpdateTransaction(Chart))
+         {
+            repositories.SelectMany(x => x.ObservationColumns()).Each(observationColumn => _editorPresenter.AddCurveForColumn(observationColumn));
+         }
       }
 
       private void addObservedDataIfNeeded(IEnumerable<DataRepository> dataRepositories)
       {
          //curves are already selected. no need to add default selection
          if (Chart.Curves.Any()) return;
-         dataRepositories.Each(addObservedData);
+         addObservedData(dataRepositories.ToList());
       }
 
-      public ICurveChart Chart
+      public CurveChart Chart
       {
-         get { return _displayPresenter.DataSource; }
-         protected set
-         {
-            if (Chart != null)
-               Chart.PropertyChanged -= onPropertyChanged;
+         get => _displayPresenter.Chart;
+         protected set => Edit(value);
+      }
 
-            _editorPresenter.DataSource = value;
-            _displayPresenter.DataSource = value;
+      public void Clear()
+      {
+         releasePropertyChanged();
 
-            if (Chart != null)
-               Chart.PropertyChanged += onPropertyChanged;
-         }
+         _editorPresenter.Clear();
+         _displayPresenter.Clear();
+      }
+
+      private void releasePropertyChanged()
+      {
+         if (Chart != null)
+            Chart.PropertyChanged -= onPropertyChanged;
+      }
+
+      private void subscribePropertyChanged()
+      {
+         if (Chart != null)
+            Chart.PropertyChanged += onPropertyChanged;
+      }
+
+      public void Edit(CurveChart curveChart)
+      {
+         releasePropertyChanged();
+
+         _editorPresenter.Edit(curveChart);
+         _displayPresenter.Edit(curveChart);
+
+         subscribePropertyChanged();
       }
 
       private void onPropertyChanged(object sender, PropertyChangedEventArgs e)
```

Comment:
There is the method ChartChanged that can be overriden in the base ChartPresenter.
If for some reason, this cannot be used, you need to react to the event ChartChanged event of the ChartEditorPresenter